### PR TITLE
Desktop notifications

### PR DESCRIPTION
**Pull request by owlnai**
Added another transport.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,344 @@
+{
```
Comment (KockaAdmiralac) [inline package-lock.json]: What is this file and shouldn't it be gitignore'd?

File: config.sample.json [hunk 2]
```diff
@@ -2,10 +2,8 @@
 	"wikis": {
 		"communitytest": {
 			"transport": {
-				"platform": "discord",
-				"id": "123456789",
-				"token": "aosBaAodia-oIA9DAkasd-9jASdlkAdu-7Askanadki-IASd8Asnak"
+				"platform": "notifications"
```
Comment (KockaAdmiralac) [inline config.sample.json]: Why was this changed?

File: transports/notifications/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+/**
```
Comment (KockaAdmiralac) [inline transports/notifications/index.js:1]: There's some inconsistent spacing in this file, please use 4 spaces instead of tabs (or at least I think other files are doing so).

File: transports/notifications/index.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+/**
+ * Importing modules
+ */
+const io = require('../../includes/io.js'),
+	util = require('../../includes/util.js'),
+	Transport = require('../transport.js'),
+	notifier = require('node-notifier'),
+	opn = require('opn');
+var config,
```
Comment (KockaAdmiralac) [inline transports/notifications/index.js]: `var` in Node🔥 👀 🔥
`config` is already `this.config` in the Notifications class and it is set after you call the constructor for the superclass
`diff` could be converted to `this.diff`, probably?

File: transports/notifications/index.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+/**
+ * Importing modules
+ */
+const io = require('../../includes/io.js'),
+	util = require('../../includes/util.js'),
+	Transport = require('../transport.js'),
+	notifier = require('node-notifier'),
+	opn = require('opn');
+let    config,
```
Comment (KockaAdmiralac) [inline transports/notifications/index.js]: I'd rather not have variables outside of the class if they aren't constants. `config` is already `this.config` inside the Notifications class, and `diff` could be converted to `this.diff`

File: transports/notifications/index.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+     * @param {String} name Template name
+     * @param {Array<String>} args Template arguments
+     * @return {String} Parsed template
+     */
+    template(name, args) {
+        switch(name) {
+            case 'diff':
+                const general = this.info.general;
+                diff = args[0];
+                console.log(diff);
```
Comment (KockaAdmiralac) [inline transports/notifications/index.js]: Don't forget to remove the debugging

File: transports/notifications/index.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                return '';
+            case 'diffSize':
+                return `(${args[0]})`;
+            case 'user':
```
Comment (KockaAdmiralac) [inline transports/notifications/index.js]: This could be
```javascript
case 'user':
case 'link':
case 'userlink':
    return args[0];
```

File: yarn.lock [hunk 8]
```diff
@@ -0,0 +1,339 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (KockaAdmiralac) [inline yarn.lock]: Yeah, looks like it should be put into .gitignore

**Review by KockaAdmiralac — CHANGES_REQUESTED** [review]
Idk if this is how GitHub pull request reviews work but the requested changes should be below

File: package-lock.json (hunk 1 above)
Comment (owlnai) [reply package-lock.json]: I don't know, it was generated when I ran "npm install"

File: config.sample.json (hunk 2 above)
Comment (owlnai) [reply config.sample.json]: Reverted, thanks

File: transports/notifications/index.js (hunk 3 above)
Comment (owlnai) [reply transports/notifications/index.js:1]: I've formatted the file again, should be fine now

File: yarn.lock (hunk 8 above)
Comment (owlnai) [reply yarn.lock]: Yep, it's auto-generated by `yarn install`

File: transports/notifications/index.js (hunk 7 above)
Comment (owlnai) [reply transports/notifications/index.js]: Done

File: transports/notifications/index.js (hunk 6 above)
Comment (owlnai) [reply transports/notifications/index.js]: Done

File: transports/notifications/index.js (hunk 5 above)
Comment (owlnai) [reply transports/notifications/index.js]: Done

**Comment by owlnai** [comment]
Kill this with fire
